src/general-redaction/rapidjsonRedactor.cpp

```diff
@@ -0,0 +1,239 @@
+#include "../../include/general-redaction/rapidjsonRedactor.hpp"
+
+RapidjsonRedactor::RapidjsonRedactor() {
+    indent = 0;
+}
+
+void RapidjsonRedactor::redactAllInstancesOfMemberByName(rapidjson::Value& value, std::string member, bool& success) {
+    if (value.IsObject()) {
+        while (value.HasMember(member.c_str())) {
+            value.RemoveMember(member.c_str());
+            success = true;
+        }
+        for (auto& m : value.GetObject()) {
+            std::string type = kTypeNames[m.value.GetType()];
+            if (type == "Object" || type == "Array") {
+                std::string name = m.name.GetString();
+                auto& v = value[name.c_str()];
+                redactAllInstancesOfMemberByName(v, member, success);
+            }
+        }
+    }
+    else if (value.IsArray()) {
+        for (auto& m : value.GetArray()) {
+            std::string type = kTypeNames[m.GetType()];
+            if (type == "Object" || type == "Array") {
+                redactAllInstancesOfMemberByName(m, member, success);
+            }
+        }
+    }
+}
+
+void RapidjsonRedactor::redactMemberByPath(rapidjson::Value& value, std::string path, bool& success) {
+    if (success) {
+        // return if redaction has already succeeded
+        return;
+    }
+    std::string nextPathElement = getTopLevelFromPath(path);
+    std::string target = getBottomLevelFromPath(path);
+
+    if (value.IsObject()) {
+        if (value.HasMember(nextPathElement.c_str())) {
+            // get the type of the next path element
+            std::string type = kTypeNames[value[nextPathElement.c_str()].GetType()];
+            if (type == "Object" || type == "Array") {
+                // if the next path element is an object or array, recurse
+                auto& v = value[nextPathElement.c_str()];
+                removeTopLevelFromPath(path);
+                redactMemberByPath(v, path, success);
+            }
+            else {
+                // if the next path element is the target, remove it
+                if (nextPathElement == target) {
+                    value.RemoveMember(nextPathElement.c_str());
+                    success = true;
+                    return;
+                }
+            }
+        }
+        else {
+            // if the next path element is not a member of the object, return
+            return;
+        }
+    }
+    else if (value.IsArray()) {
+        for (auto& m : value.GetArray()) {
+            std::string type = kTypeNames[m.GetType()];
+            if (type == "Object" || type == "Array") {
+                redactMemberByPath(m, path, success);
+            }
+        }
+    }
+}
+
+void RapidjsonRedactor::searchForMemberByName(rapidjson::Value& value, std::string member, bool& success) {
+    if (success) {
```

Comment:
Same refactor comment, tweak to return boolean instead of ref?

Reply:
Tweaked!